buyer.go

```diff
@@ -146,18 +152,48 @@ func newTicketPurchaser(cfg *config,
 		priceMode = AvgPriceDualMode
 	}
 
+	// Determine if there have been existing runs and use buy period
+	f, err := os.Open(filepath.Join(csvPath, csvPurchasedFn))
+	if err != nil {
+		log.Errorf("Error opening file: %v", err)
+		return nil, err
+	}
+	defer f.Close()
+
+	rdr := csv.NewReader(f)
+	rdr.Comma = ','
+	prevRecord := []string{}
+	for {
+		record, err := rdr.Read()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+		}
+		prevRecord = record
+	}
+	prevToBuyDiffPeriod := 0
+	if len(prevRecord) >= 4 {
```

Comment:
You need to check the height here as well. If this is outside the last difficulty window, the value should be ignored and the number of tickets to buy recalculated from scratch.